algebra-core/src/fields/macros.rs

```diff
@@ -274,7 +281,18 @@ macro_rules! impl_Fp {
         impl<P: $FpParameters> FromBytes for $Fp<P> {
             #[inline]
             fn read<R: Read>(reader: R) -> IoResult<Self> {
-                BigInteger::read(reader).map($Fp::from_repr)
+                BigInteger::read(reader).and_then( |b|
+                    if b.is_zero() {
+                        Ok($Fp::zero())
+                    } else {
+                        let f = $Fp::from_repr(b);
+                        if f == $Fp::zero() {
```

Comment:
This is nice, it checks if the BigNumber is 0, and returns Fp::zero, and in the else branch, if the Fp::from_repr returned zero, it means that there was an error during reading

Reply:
Yep! Should definitely think about improving `from_repr` though